src/io/fastq.rs

```diff
@@ -3,24 +3,103 @@
 // This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! FastQ reading.
+//! Structs and trait to read and write file in FASTA format.
 //!
 //! # Example
 //!
+//! ## Read
+//!
+//! In this example we parse fasta file from stdin and compute some statistics
+//!
 //! ```
+//! use std::io;
 //! use bio::io::fastq;
-//! use bio::io::fastq::FastqRead;
+//! let mut reader = fastq::Reader::new(io::stdin());
+//!
+//! let mut nb_reads = 0;
+//! let mut nb_bases = 0;
+//!
+//! for result in reader.records() {
+//!     let record = result.expect("Error durring fasta record parsing");
+//!
+//!     nb_reads += 1;
+//!     nb_bases += record.seq().len();
+//! }
+//!
+//! println!("Number of reads: {}", nb_reads);
+//! println!("Number of bases: {}", nb_bases);
+//! ```
+//!
+//! We can also use a `while` loop to iterate over record
+//! ```
 //! use std::io;
+//! use bio::io::fastq;
+//! let mut records = fastq::Reader::new(io::stdin()).records();
+//!
+//! let mut nb_reads = 0;
+//! let mut nb_bases = 0;
+//!
+//! while let Some(Ok(record)) = records.next() {
+//!     nb_reads += 1;
+//!     nb_bases += record.seq().len();
+//! }
+//!
+//! println!("Number of reads: {}", nb_reads);
+//! println!("Number of bases: {}", nb_bases);
+//! ```
+//!
+//! ## Write
+//!
+//! In this example we generate 10 random sequence with length 100 and write it in stdout.
+//!
+//! ```
+//! use std::io;
+//! use bio::io::fastq;
+//!
+//! let mut seed = 42;
+//!
+//! let nucleotides = [b'A', b'C', b'G', b'T'];
+//!
+//! let mut writer = fastq::Writer::new(io::stdout());
+//!
+//! for _ in 0..10 {
+//!     let seq = (0..100).map(|_| {
+//!         seed = ((seed ^ seed << 13) ^ seed >> 7) ^ seed << 17; // don't use this random generator
+//!         nucleotides[seed % 4]
+//!     }).collect::<Vec<u8>>();
+//!
+//!     let qual = (0..100).map(|_| b'!').collect::<Vec<u8>>();
+//!
+//!    writer.write("random", None, seq.as_slice(), qual.as_slice());
+//! }
+//! ```
+//!
+//! ## Read and Write
+//!
+//! In this example we filter reads from stdin on mean quality (Phred + 33) and write it in stdout
```

Comment:
write them to stdout

Reply:
btw, you can use codeblocks with "suggestion" after the first three backticks for suggestions, like this:
```suggestion
//! In this example we filter reads from stdin on mean quality (Phred + 33) and write them to stdout
```

Reply:
@tedil Thanks, I like this feature of GitHub. I've gone back to the terminal to correct the rest by `sed` as they seem to be systematic (repeated) issues.

Reply:
Wow, that is awesome!! Thanks @tedil